src/transformers/models/granite_speech/feature_extraction_granite_speech.py

```diff
@@ -178,11 +176,8 @@ def _get_audios_and_audio_lengths(self, audios: AudioInput) -> Sequence["torch.T
             if not torch.is_floating_point(audios[0]):
                 raise ValueError("Invalid audio provided. Audio should be a floating point between 0 and 1")
             lengths = [audio.shape[-1] for audio in audios]
-            padding = [max(lengths) - length for length in lengths]
-            # ensure all audios have a batch dimension:
-            audios = [audio.view(1, -1) for audio in audios]
-            padded = [torch.nn.functional.pad(audio, (0, pad)) for audio, pad in zip(audios, padding)]
-            audios = torch.cat(padded, dim=0)
+            audios = [audio.squeeze(0) for audio in audios]
```

Comment:
`pad_sequence` is much faster

Reply:
wow that's good to know! How much faster is it exactly?

Reply:
@ArthurZucker I was surprised too. 

I think there are two reasons for that:
- The previous code required multiple memory allocations. (Each padding requires a new memory allocation, as well as the concatenation.)
- `pad_sequence` is compiled. 

I ran it for 500 repeats:
```
old took 2.545 seconds
new took 1.096 seconds
```

It gave large utilization improvements when using the model in `trl`.
Also, the code looks cleaner. :)